src/main/java/com/csforge/sstable/Query.java

```diff
@@ -0,0 +1,178 @@
+import org.apache.cassandra.exceptions.InvalidRequestException;
+import org.apache.cassandra.io.sstable.Descriptor;
+import org.apache.cassandra.io.sstable.format.SSTableReader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.mockito.Mockito.*;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.*;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+public class Query {
+    private static final Logger logger = LoggerFactory.getLogger(Query.class);
+
+    public static QueryOptions OPTIONS = QueryOptions.DEFAULT;
+
+    static {
+        Config.setClientMode(true);
+
+        // Partitioner is not set in client mode.
+        if (DatabaseDescriptor.getPartitioner() == null)
+            DatabaseDescriptor.setPartitionerUnsafe(Murmur3Partitioner.instance);
+    }
+
+    StatementRestrictions restrictions = null;
+    SelectStatement.RawStatement statement = null;
+    CFMetaData cfm = null;
+    Selection selection = null;
+    ColumnFilter queriedColumns;
+    String path;
+
+    public Query(String query, CFMetaData cfm) throws IllegalAccessException, NoSuchFieldException, IOException {
+        SelectStatement.RawStatement statement = (SelectStatement.RawStatement) QueryProcessor.parseStatement(query);
+        if (!statement.parameters.orderings.isEmpty()) {
+            throw new UnsupportedOperationException("ORDER BY not supported");
+        }
+        if (statement.parameters.isDistinct) {
+            throw new UnsupportedOperationException("DISTINCT not supported");
+        }
+        this.path = statement.columnFamily();
+
+        VariableSpecifications boundNames = statement.getBoundVariables();
+
+        Selection selection = statement.selectClause.isEmpty()
+                ? Selection.wildcard(cfm)
+                : Selection.fromSelectors(cfm, statement.selectClause);
+
+        // yes its unfortunate, im sorry
+        StatementType type = mock(StatementType.class);
+        when(type.allowClusteringColumnSlices()).thenReturn(true);
+        when(type.allowNonPrimaryKeyInWhereClause()).thenReturn(true);
+        when(type.allowPartitionKeyRanges()).thenReturn(true);
+        when(type.allowUseOfSecondaryIndices()).thenReturn(false);
+
+        StatementRestrictions restrictions = new StatementRestrictions(
+                type,
+                cfm,
+                statement.whereClause,
+                boundNames,
+                selection.containsOnlyStaticColumns(),
+                selection.containsACollection(),
+                true, true);
+
+        this.restrictions = restrictions;
+        this.statement = statement;
+        this.cfm = cfm;
+        this.selection = selection;
+
+        ColumnFilter filter;
+        if (selection.isWildcard()) {
+            filter = ColumnFilter.all(cfm);
+        } else {
+            ColumnFilter.Builder builder = ColumnFilter.allColumnsBuilder(cfm);
```

Comment:
This doesn't seem to do column filters correctly because the builder includes all columns, the following does seem to work though:

```java
        ColumnFilter filter;
        if (selection.isWildcard()) {
            filter = ColumnFilter.all(cfm);
        } else {
            ColumnFilter.Builder builder = ColumnFilter.selectionBuilder();
            selection.getColumns().stream().filter(def -> !def.isPrimaryKeyColumn()).forEach(builder::add);
            //builder.addAll(restrictions.nonPKRestrictedColumns(true));
            filter = builder.build();
        }
```